File: problem010.py
# Find the sum of all the primes below two million.

from math import floor, sqrt


def find_primes_eratosthenes(n):
    is_prime = [True] * n
    for potential_prime in range(2, floor(sqrt(n))):
        if is_prime[potential_prime]:
            for multiple in range(potential_prime**2, n, potential_prime):
                is_prime[multiple] = False
    return [index for index, val in enumerate(is_prime) if val and index > 1]


print(sum(find_primes_eratosthenes(2_000_000)))

File: c/problem010.c
// Find the sum of all the primes below two million.
#include <math.h>
#include <stdio.h>

// Using the sieve of eratosthenes
long findSumOfPrimesUpTo(int maxValue) {
  long primeSum = 0;
  int isPrime[maxValue];
  for (int i = 0; i < maxValue; i++) {isPrime[i] = 1;}

  for (int i = 2; i <= sqrt(maxValue); i++) {
    if (isPrime[i]) {
      for (int multiple = pow(i, 2); multiple < maxValue; multiple += i) {
        isPrime[multiple] = 0;
      }
    }
  }

  for (int i = 2; i < maxValue; i++) { primeSum += isPrime[i] ? i : 0; }

  return primeSum;
}

int main() {
  printf("%lu\n", findSumOfPrimesUpTo(2000000));
  return 0;
}
